src/new-components/notifications/notifications.tsx

```diff
@@ -0,0 +1,238 @@
+     * You can even pass `false` if you want a certain notification to never be dismissed
+     * automatically.
+     *
+     * @default 10 (seconds)
+     */
+    defaultAutoDismissDelay: number
+}
+
+const NotificationContext = React.createContext<NotificationActions>({
+    showNotification: () => undefined,
+    dismissNotification: () => undefined,
+})
+
+type NotificationsProviderProps = Partial<NotificationsConfig> & {
+    children: NonNullable<React.ReactNode>
+    padding?: Space
+}
+
+/**
+ * Provides the state management and rendering of the notifications currently active.
+ *
+ * You need to render this near the top of your app components tree, in order to `useNotifications`.
+ *
+ * @see useNotifications
+ */
+function NotificationsProvider({
+    children,
+    padding = 'large',
+    defaultAutoDismissDelay = 10 /* seconds */,
+    defaultDismissLabel = 'Close',
+}: NotificationsProviderProps) {
+    const [notifications, setNotifications] = React.useState<NotificationsList>([])
+
+    const value: NotificationActions = React.useMemo(
+        () => ({
+            showNotification(props) {
+                const notificationId = generateElementId('notification')
+                const newNotification: InternalNotificationProps = {
+                    autoDismissDelay: defaultAutoDismissDelay,
+                    dismissLabel: defaultDismissLabel,
+                    ...props,
+                    notificationId,
+                }
+                setNotifications((list) => [...list, newNotification])
+            },
+
+            dismissNotification(notificationId) {
+                setNotifications((list) => {
+                    const index = list.findIndex((n) => n.notificationId === notificationId)
+                    if (index < 0) return list
+                    const copy = [...list]
+                    copy.splice(index, 1)
+                    return copy
+                })
+            },
+        }),
+        [defaultDismissLabel, defaultAutoDismissDelay],
+    )
+
+    return (
+        <NotificationContext.Provider value={value}>
+            {children}
+            <Portal>
+                <StackedNotificationsView notifications={notifications} padding={padding} />
+            </Portal>
+        </NotificationContext.Provider>
+    )
+}
+
+/** @private */
+function StackedNotificationsView({
+    notifications,
+    padding,
+}: {
+    notifications: NotificationsList
+    padding: Space
+}) {
+    return notifications.length === 0 ? null : (
+        <Box
+            className={styles.stackedView}
+            position="fixed"
+            width="full"
+            paddingLeft={padding}
+            paddingBottom={padding}
+        >
+            <Stack space="medium">
+                {notifications.map(({ notificationId, ...props }) => (
+                    <Notification key={notificationId} notificationId={notificationId} {...props} />
```

Comment:
Not sure if design has planned for it yet, but I think we should have a subtle enter/exit transition for these, otherwise it feels a bit abrupt.

Reply:
Sure, I thought about it. But I could not justify devoting time for this right now. It'll have to be for a subsequent version.

Reply:
I'm doing something about this. Stay tuned…

Reply:
Check out the demo section in the PR description.